### PR TITLE
Fix tests CPU

### PR DESCRIPTION
**Pull request by tkornuta-nvidia**
- [x] All tests running on CPUs (aside of those indicated as running only on GPU;))

Refactor of export deploy tests:
- [x] All modules are loaded created on the basis of a configuration file
- [x] Use `tmpdir` pytest fixture for creation of temporary files
- [x] Export tests parametrized by `input_example`, `module_name`(type) and `df_type` (export type) (thanks @ryanleary for laying the foundations and motivating me to push it further!)

Configuration import/export tests:
- [x] Use tmpdir pytest fixture for creation of temporary files

Jenkins:
- [x] Added Unit, Integration and System tests on CPU to commits to every opened PR - along the line of:
https://docs.google.com/presentation/d/1i-aaCTCijBt9N_NPjlFcCYpDiCA48jYtZ22OIJPNLAY/edit#slide=id.p1

Additional:
- [x] Removed "unformatting" of `test_deploy_export.py` and `tensorrt_runner.py` files (works!)

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -33,5 +33,4 @@ markers =
 known_localfolder = nemo,tests
 sections = FUTURE,STDLIB,THIRDPARTY,LOCALFOLDER
 default_section = THIRDPARTY
-#TODO tests/unit/core/test_deploy_export.py gets screwed by isort
-skip = setup.py, nemo/utils/__init__.py, tests/unit/core/test_deploy_export.py
```
Comment (okuchaiev) [inline setup.cfg]: please put this back - there are some issues around black/isort on some of dev machines which required us to add skip test_deploy_export

Comment (tkornuta-nvidia) [reply setup.cfg]: ok

File: tests/unclassified/test_unclassified_tts.py [hunk 2]
```diff
@@ -280,7 +276,8 @@ def test_fastspeech(self):
         callback = nemo.core.SimpleLossLoggerCallback(
             tensors=[loss_t], print_func=lambda x: logging.info(f'Train Loss: {str(x[0].item())}'),
         )
-        optimizer = neural_factory.get_trainer()
+        # Instantiate an optimizer to perform `train` action
+        optimizer = nemo.backends.pytorch.actions.PtActions()
         optimizer.train(
```
Comment (blisc) [inline tests/unclassified/test_unclassified_tts.py]: Shouldn't this be nf.train()?

File: tests/unit/core/test_module_configuration.py [hunk 3]
```diff
@@ -59,7 +59,7 @@ def test_build_in_types(self):
         params = {"int": 123, "float": 12.4, "string": "ala ma kota", "bool": True}
 
         # Check error output.
-        self.assertEqual(self.module.validate_params(params), True)
+        assert self.module.validate_params(params) == True
```
Comment (blisc) [inline tests/unit/core/test_module_configuration.py]: Is there a reason we are switching this?

**Comment by blisc** [comment]
So I missed the other PRs on the export tests but I currently see this:
`tests/unit/core/test_deploy_export.py::TestDeployExport::test_module_export[input_example4-TokenClassifier-DeploymentFormat.TRTONNX] SKIPPED [ 31%]`
Do we want to skip this?

File: tests/unclassified/test_unclassified_tts.py (hunk 2 above)
Comment (tkornuta-nvidia) [reply tests/unclassified/test_unclassified_tts.py]: I have copied that from one of the existing examples a long time ago. This is the call that I was consistently using along all tests/examples that I have refactored. 

So when  I was removing the local instance of NF I did the same here. At the end it is doing the same as self.nf.train(), i.e. creating action instance.

File: tests/unit/core/test_module_configuration.py (hunk 3 above)
Comment (tkornuta-nvidia) [reply tests/unit/core/test_module_configuration.py]: Yes, please note that this test is not derived from unittest TestCase, so we do not have the access to TestCase assert methods.

This is one step towards more tight dependency on pytest, in here I did that as I started to rely on pytest 'tmpdir' fixture... honestly speaking now I am not not even sure if it was failing or not, not important I guess... the result is the same.

**Comment by tkornuta-nvidia** [comment]
Replying to @blisc comment:
https://github.com/NVIDIA/NeMo/pull/503#issuecomment-601432396

Exactly that was the goal of the previous @ryanleary PR #501 , where he wanted to test some exports even when one does not have TensorRT installed. From "table driven testing" point of view  that makes perfectly sense.

Now I have pushed that further, and implemented a single test parametrized by not only export type, but also module (plus input data, but it is a by-product of parametrization with module).

File: tests/unclassified/test_unclassified_tts.py (hunk 2 above)
Comment (blisc) [reply tests/unclassified/test_unclassified_tts.py]: But this is against the nemo design pattern.
If you think this is outside the scope of PR, that's fine, but we should make a note to change then back to self.nf...

**Comment by blisc** [comment]
> Replying to @blisc comment:
> [#503 (comment)](https://github.com/NVIDIA/NeMo/pull/503#issuecomment-601432396)
> 
> Exactly that was the goal of the previous @ryanleary PR #501 , where he wanted to test some exports even when one does not have TensorRT installed. From "table driven testing" point of view that makes perfectly sense.
> 
> Now I have pushed that further, and implemented a single test parametrized by not only export type, but also module (plus input data, but it is a by-product of parametrization with module).

Yes, but our CI pipeline should have TRT installed and tested. Users can do whatever they want, but our official tests should not be skipping these.

**Comment by tkornuta-nvidia** [comment]
Ok, now I got your point! It seems that our CI is also skipping the TRT tests!
http://10.110.42.104:8080/blue/organizations/jenkins/NeMo/detail/PR-503/7/pipeline
```
tests/unit/core/test_deploy_export.py::TestDeployExport::test_module_export[input_example4-TokenClassifier-DeploymentFormat.TRTONNX] SKIPPED [ 31%]
```
Hmmm, seems that I have messed up the parametrization construction, let me try to fix that. Thanks @blisc for catching this!

**Comment by tkornuta-nvidia** [comment]
@okuchaiev @ryanleary has anything changed in our CI configuration? 

In particular asking about TRT, as TRT tests are skipped:
http://10.110.42.104:8080/blue/organizations/jenkins/NeMo/detail/PR-503/7/pipeline

 I have double checked that the construct:
```
            pytest.param(torch.randn(16, 16, 512).cuda(), "TokenClassifier", DF.TRTONNX, marks=requires_trt),
```
works properly. I have also checked the logic behind `requires_trt` mark, looks good to me (please note that I haven't touched it in this PR at all).

Still, the TRT tests are skipped. Any ideas?

**Comment by tkornuta-nvidia** [comment]
@blisc @okuchaiev @ryanleary 
I have added a simple print in exception handing section. Currently our CI prints:
```
collecting ... [NeMo E 2020-03-20 20:03:34 test_deploy_export:68] Failed to import: `attempted relative import with no known parent package` (<class 'ImportError'>)
```
This clearly indicates that the problem is not caused by my changes, but import failure - thus something changed in the configuration of CI machine, probably with PyCuda or TensorRT.

Do you guys agree that fixing this  is out of the scope of this PR?

**Comment by blisc** [comment]
> @blisc @okuchaiev @ryanleary
> I have added a simple print in exception handing section. Currently our CI prints:
> 
> ```
> collecting ... [NeMo E 2020-03-20 20:03:34 test_deploy_export:68] Failed to import: `attempted relative import with no known parent package` (<class 'ImportError'>)
> ```
> 
> This clearly indicates that the problem is not caused by my changes, but import failure - thus something changed in the configuration of CI machine, probably with PyCuda or TensorRT.
> 
> Do you guys agree that fixing this is out of the scope of this PR?

So let me ask you this question: Why did you delete the __init__.py files in the test folders?

**Comment by ryanleary** [comment]
Removing all the init.py files breaks relative imports. Put them back and
this will work.

And in general I disagree that fixing this is out of scope. A restructuring
that introduced regressions shouldn’t be accepted.
-- 

Ryan Leary

**Comment by tkornuta-nvidia** [comment]
I have removed `__init__` as from:

https://docs.pytest.org/en/latest/goodpractices.html

I have understood that in case of tests those are _unnecessary_, and should be added only if:

"If you need to have test modules with the same name, you might add __init__.py files to your tests folder and subfolders, changing them to packages:"

Seems that in this case I was wrong - we have additional files that are imported that will require __init__ files. Hopefully it will work. Let me fix the fix the fix.

Thanks for the help, guys!

**Comment by tkornuta-nvidia** [comment]
Ps. Please note that I have removed all init.py files from tests/ directories already some time ago, so thought that someone added them by an accident... :]

**Comment by tkornuta-nvidia** [comment]
Ok, this really was the cause! 

```test_module_export_input_examp3/export/TokenClassifier.trt.onnx PASSED```

Reverting the print flag, and I think we are ready to go.

**Comment by tkornuta-nvidia** [comment]
```
tests/unit/core/test_deploy_export.py::TestDeployExport::test_module_export[input_example13-QuartznetEncoder-DeploymentFormat.ONNX] PASSED [ 42%]
tests/unit/core/test_deploy_export.py::TestDeployExport::test_module_export[input_example14-QuartznetEncoder-DeploymentFormat.TORCHSCRIPT] PASSED [ 43%]
tests/unit/core/test_deploy_export.py::TestDeployExport::test_module_export[input_example15-QuartznetEncoder-DeploymentFormat.PYTORCH] PASSED [ 44%]
tests/unit/core/test_deploy_export.py::TestDeployExport::test_module_export[input_example16-QuartznetEncoder-DeploymentFormat.TRTONNX] PASSED [ 45%]
```

**Pull request merged**